### PR TITLE
Slider: add range support

### PR DESCRIPTION
**Pull request by Leopoldthecoder**
Please makes sure these boxes are checked before submitting your PR, thank you!

* [ ] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md)).
* [ ] Make sure you are merging your commits to `dev` branch.
* [ ] Add some descriptions and refer relative issues for you PR.

#1667

### CODE REVIEW

File: packages/slider/src/main.vue [hunk 1]
```diff
@@ -79,142 +82,137 @@
...
-        oldValue: this.value,
-        currentPosition: (this.value - this.min) / (this.max - this.min) * 100 + '%'
+        dragging: false
       };
     },
 
     watch: {
       inputValue(val) {
-        this.$emit('input', Number(val));
+        this.value1 = val;
```
Comment (QingWei-Li) [inline packages/slider/src/main.vue]: 如此洒脱的命名

File: packages/slider/src/main.vue [hunk 2]
```diff
@@ -79,142 +82,137 @@
...
-        let value = steps * lengthPerStep * (this.max - this.min) * 0.01 + this.min;
-        value = parseFloat(value.toFixed(this.precision));
-        this.$emit('input', value);
-        this.currentPosition = (this.value - this.min) / (this.max - this.min) * 100 + '%';
-        if (!this.dragging) {
-          if (this.value !== this.oldValue) {
-            this.$emit('change', this.value);
-            this.oldValue = this.value;
-          }
+      value2() {
```
Comment (QingWei-Li) [inline packages/slider/src/main.vue]: 如此洒脱的命名2

File: packages/slider/src/main.vue (hunk 1 above)
Comment (Leopoldthecoder) [inline packages/slider/src/main.vue]: 不洒脱了。

File: packages/slider/src/main.vue (hunk 2 above)
Comment (Kingwl) [inline packages/slider/src/main.vue]: 23333

File: packages/slider/src/main.vue [hunk 3]
```diff
@@ -79,142 +82,136 @@
...
-          this.newPos = this.startPos + diff;
-          this.setPosition(this.newPos);
+      setValues() {
+        const val = this.value;
+        if (this.range && Array.isArray(val)) {
+          if (val[1] < this.min) {
+            this.$emit('input', [this.min, this.min]);
+          } else if (val[0] > this.max) {
+            this.$emit('input', [this.max, this.max]);
+          } else if (val[0] < this.min) {
```
Comment (baiyaaaaa) [inline packages/slider/src/main.vue]: 减少两个if else
else {
  this.firstValue = Math.max(val[0], this.min);
  this.secondValue = Math.min(val[1], this.max);
  this.$emit('input', this.firstValue, this.secondValue]);
}

File: packages/slider/src/main.vue [hunk 4]
```diff
@@ -79,142 +82,136 @@
...
+            if (this.valueChanged()) {
+              this.$emit('change', [this.minValue, this.maxValue]);
+              this.oldValue = val.slice();
+            }
+          }
+        } else if (!this.range && typeof val === 'number' && !isNaN(val)) {
+          if (val < this.min) {
+            this.$emit('input', this.min);
+          } else if (val > this.max) {
+            this.$emit('input', this.max);
```
Comment (baiyaaaaa) [inline packages/slider/src/main.vue]: this.$emit('input', Math.min(val, this.max));

**Pull request merged**